src/main/java/org/lsc/Hooks.java

```diff
@@ -0,0 +1,189 @@
+import java.io.StringWriter;
+import java.io.PrintWriter;
+import org.lsc.utils.output.LdifLayout;
+import com.fasterxml.jackson.databind.ObjectMapper; // For encoding object to JSON
+import com.fasterxml.jackson.databind.ObjectWriter;
+
+
+/**
+ * This object is managing posthook scripts
+ */
+public class Hooks {
+
+	static final Logger LOGGER = LoggerFactory.getLogger(AbstractSynchronize.class);
+	/**
+	 * Method calling a postSyncHook if necessary
+	 *
+	 * return nothing
+	 */
+	public final static void postSyncHook(final String hook, final String outputFormat, final LscModifications lm) {
+
+		if( hook != null && ! hook.equals("") )
+		{
+
+			String format = "";
+			if( outputFormat.equals("json") ) {
+				format = "json";
+			}
+			else
+			{
+				format = "ldif";
+			}
+
+			// Compute json/ldif modifications
+			String modifications = null;
+
+			switch (lm.getOperation()) {
+				case CREATE_OBJECT:
+					if( format.equals("json") ) {
+						modifications = getJsonModifications(lm);
+					}
+					else {
+						modifications = LdifLayout.format(lm);
+					}
+					callHook("create", hook, lm.getMainIdentifier(), format, modifications);
+					break;
+
+				case UPDATE_OBJECT:
+					if( format.equals("json") ) {
+						modifications = getJsonModifications(lm);
+					}
+					else {
+						modifications = LdifLayout.format(lm);
+					}
+					callHook("update", hook, lm.getMainIdentifier(), format, modifications);
+					break;
+
+				case CHANGE_ID:
+					if( format.equals("json") ) {
+						modifications = getJsonModifications(lm);
+					}
+					else {
+						modifications = LdifLayout.format(lm);
+					}
+					callHook("changeId", hook, lm.getMainIdentifier(), format, modifications);
+					break;
+
+				case DELETE_OBJECT:
+					callHook("delete", hook, lm.getMainIdentifier(), format, modifications);
+					break;
+
+				default:
+					LOGGER.info("Error: unknown operation for posthook {}", hook);
+			}
+		}
+	}
+
+	/**
+	* Method calling the hook
+	*
+	* return nothing
+	*/
+	public final static void callHook(	String operationType,
+						String hook,
+						String identifier,
+						String format,
+						String modifications) {
+
+		LOGGER.info("Calling {} posthook {} with format {} for {}", operationType, hook, format, identifier);
+		try {
+			if( modifications != null ) {
+				Process p = new ProcessBuilder(
+					hook,
+					identifier,
+					operationType)
+				.start();
+
+				// sends ldif modifications to stdin of hook script
+				OutputStream stdin = p.getOutputStream();
+				stdin.write(modifications.getBytes());
+				stdin.write("\n".getBytes());
+				stdin.flush();
+				stdin.close();
+			}
+			else {
+				Process p = new ProcessBuilder(
+					hook,
+					identifier,
+					operationType)
+				.start();
+			}
+		}
+		catch(IOException e) {
+			StringWriter sw = new StringWriter();
+			PrintWriter pw = new PrintWriter(sw);
+			e.printStackTrace(pw);
+			LOGGER.error("Error while calling {} posthook {} with format {} for {}: {}",
```

Comment:
do you know that the LOGGER.error can directly take an exception as a parameter, and so log the stack trace without all this stuff?